### PR TITLE
feat(discover) Improve rendering of array fields

### PR DESCRIPTION
**Pull request by markstory**
Add a toggle that lets users expand array fields and see all the values. The toggle follows the 'last' item which is the item we show when the field is collapsed. This will be applied to the various `stack.*` fields and fields like `error.type` and `error.value`.

Screen Shot 2020-11-04 at 1 56 06 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/discover/arrayValue.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return (
+      <ArrayContainer>
+        {expanded &&
+          value
+            .slice(0, value.length - 1)
+            .map(item => <ArrayItem key={item}>{item}</ArrayItem>)}
+        <span>
+          {value.slice(-1)[0]}
+          <button onClick={this.handleToggle}>
+            {expanded ? t('[collapse]') : t('[+%s more]', value.length - 1)}
```
Comment (wmak) [inline src/sentry/static/sentry/app/utils/discover/arrayValue.tsx:37]: When `value` is an empty array this is rendering as `[+-1 more]`

File: src/sentry/static/sentry/app/utils/discover/arrayValue.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  render() {
+    const {expanded} = this.state;
+    const {value} = this.props;
+    return (
+      <ArrayContainer>
+        {expanded &&
+          value
+            .slice(0, value.length - 1)
+            .map(item => <ArrayItem key={item}>{item}</ArrayItem>)}
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/discover/arrayValue.tsx:33]: Will item here always be unique to be used as a key? I believe that it's possible for two items to be equal for some array fields.

Comment (markstory) [reply src/sentry/static/sentry/app/utils/discover/arrayValue.tsx]: Good point, I'll use the `item,index` instead.

File: src/sentry/static/sentry/app/utils/discover/arrayValue.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/utils/discover/arrayValue.tsx]: Fixed. I added tests for this component as it got past my 'simple' threshold.

**Review by wmak — APPROVED** [review]
🥳 This is a nice solution for array fields

**Pull request merged**